Cannot build from source

```
cc@pretty:~$ CARGO_TARGET_DIR=/tmp/cargo-installzFCk1C  cargo install --git https://github.com/orhun/halp --locked
    Updating git repository `https://github.com/orhun/halp`
  Installing halp v0.2.0 (https://github.com/orhun/halp#17161bbb)
    Updating crates.io index
warning: package `subtle v2.6.0` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
  Downloaded toml v0.9.4
  Downloaded clap_complete v4.5.55
  Downloaded thiserror-impl v1.0.68
  Downloaded httparse v1.10.0
  Downloaded thiserror v1.0.68
  Downloaded cc v1.2.11
  Downloaded terminal_size v0.4.0
  Downloaded attr_alias v0.1.3
  Downloaded subtle v2.6.0
  Downloaded roff v0.2.1
  Downloaded process_control v5.1.1
  Downloaded 11 crates (353.9KiB) in 0.77s
   Compiling proc-macro2 v1.0.85
   Compiling unicode-ident v1.0.12
   Compiling libc v0.2.155
   Compiling cfg-if v1.0.0
   Compiling rustix v0.38.34
   Compiling linux-raw-sys v0.4.14
   Compiling bitflags v2.5.0
   Compiling anstyle-query v1.1.0
   Compiling colorchoice v1.0.1
   Compiling anstyle v1.0.8
   Compiling serde v1.0.219
   Compiling once_cell v1.19.0
   Compiling rustls-pki-types v1.11.0
   Compiling is_terminal_polyfill v1.70.0
   Compiling spin v0.9.8
   Compiling cc v1.2.11
   Compiling anstyle-parse v0.2.4
   Compiling httparse v1.10.0
   Compiling clap_lex v0.7.4
   Compiling log v0.4.25
   Compiling unicode-width v0.2.0
   Compiling thiserror v1.0.68
   Compiling adler v1.0.2
   Compiling rustls v0.23.23
   Compiling itoa v1.0.14
   Compiling bytes v1.9.0
   Compiling crc32fast v1.4.2
   Compiling winnow v0.7.12
   Compiling subtle v2.6.0
   Compiling thiserror v2.0.12
   Compiling percent-encoding v2.3.1
   Compiling attr_alias v0.1.3
   Compiling toml_writer v1.0.2
   Compiling anstream v0.6.14
   Compiling roff v0.2.1
   Compiling webpki-roots v0.26.8
   Compiling rustls-pemfile v2.2.0
   Compiling miniz_oxide v0.7.4
   Compiling getrandom v0.2.15
   Compiling console v0.15.11
   Compiling dirs-sys v0.5.0
   Compiling ring v0.17.8
   Compiling process_control v5.1.1
   Compiling http v1.2.0
   Compiling console v0.16.0
   Compiling dirs v6.0.0
   Compiling quote v1.0.36
   Compiling flate2 v1.0.30
   Compiling terminal_size v0.4.0
   Compiling toml_parser v1.0.1
   Compiling ureq-proto v0.4.2
   Compiling syn v2.0.87
   Compiling clap_builder v4.5.42
   Compiling serde_derive v1.0.219
   Compiling clap_derive v4.5.41
   Compiling thiserror-impl v1.0.68
   Compiling thiserror-impl v2.0.12
   Compiling clap v4.5.42
   Compiling dialoguer v0.11.0
   Compiling serde_spanned v1.0.0
   Compiling toml_datetime v0.7.0
   Compiling clap_complete v4.5.55
   Compiling clap_mangen v0.2.29
   Compiling toml v0.9.4
   Compiling rustls-webpki v0.102.8
   Compiling ureq v3.0.12
   Compiling halp v0.2.0 (/home/cc/.cargo/git/checkouts/halp-5b2a93481a690208/17161bb)
warning: ignoring -C extra-filename flag due to -o flag

error[E0432]: unresolved import `ureq::AgentBuilder`
 --> src/helper/docs/cheat_sh.rs:3:12
  |
3 | use ureq::{AgentBuilder, Request};
  |            ^^^^^^^^^^^^ no `AgentBuilder` in the root

error[E0432]: unresolved import `ureq::AgentBuilder`
 --> src/helper/docs/cheatsheets.rs:2:12
  |
2 | use ureq::{AgentBuilder, Request};
  |            ^^^^^^^^^^^^ no `AgentBuilder` in the root

error[E0432]: unresolved import `ureq::AgentBuilder`
 --> src/helper/docs/eg.rs:2:12
  |
2 | use ureq::{AgentBuilder, Request};
  |            ^^^^^^^^^^^^ no `AgentBuilder` in the root

error[E0433]: failed to resolve: could not find `ErrorKind` in `ureq`
  --> src/helper/docs/mod.rs:64:30
   |
64 |         if e.kind() == ureq::ErrorKind::HTTP {
   |                              ^^^^^^^^^ could not find `ErrorKind` in `ureq`
   |
help: consider importing one of these enums
   |
11 + use std::io::ErrorKind;
   |
11 + use clap::error::ErrorKind;
   |
help: if you import `ErrorKind`, refer to it directly
   |
64 -         if e.kind() == ureq::ErrorKind::HTTP {
64 +         if e.kind() == ErrorKind::HTTP {
   |

error[E0603]: struct `Request` is private
   --> src/helper/docs/cheat_sh.rs:3:26
    |
  3 | use ureq::{AgentBuilder, Request};
    |                          ^^^^^^^ private struct
    |
note: the struct `Request` is defined here
   --> /home/cc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ureq-3.0.12/src/lib.rs:524:12
    |
524 | use http::{Request, Response, Uri};
    |            ^^^^^^^
help: import `Request` directly
    |
  3 | use ureq::{AgentBuilder, http::request::Request};
    |                          +++++++++++++++

error[E0603]: struct `Request` is private
   --> src/helper/docs/cheatsheets.rs:2:26
    |
  2 | use ureq::{AgentBuilder, Request};
    |                          ^^^^^^^ private struct
    |
note: the struct `Request` is defined here
   --> /home/cc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ureq-3.0.12/src/lib.rs:524:12
    |
524 | use http::{Request, Response, Uri};
    |            ^^^^^^^
help: import `Request` directly
    |
  2 | use ureq::{AgentBuilder, http::request::Request};
    |                          +++++++++++++++

error[E0603]: struct `Request` is private
   --> src/helper/docs/eg.rs:2:26
    |
  2 | use ureq::{AgentBuilder, Request};
    |                          ^^^^^^^ private struct
    |
note: the struct `Request` is defined here
   --> /home/cc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ureq-3.0.12/src/lib.rs:524:12
    |
524 | use http::{Request, Response, Uri};
    |            ^^^^^^^
help: import `Request` directly
    |
  2 | use ureq::{AgentBuilder, http::request::Request};
    |                          +++++++++++++++

error[E0603]: struct `Request` is private
   --> src/helper/docs/mod.rs:22:11
    |
 22 | use ureq::Request;
    |           ^^^^^^^ private struct
    |
note: the struct `Request` is defined here
   --> /home/cc/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ureq-3.0.12/src/lib.rs:524:12
    |
524 | use http::{Request, Response, Uri};
    |            ^^^^^^^
help: import `Request` directly
    |
 22 - use ureq::Request;
 22 + use http::request::Request;
    |

Some errors have detailed explanations: E0432, E0433, E0603.
For more information about an error, try `rustc --explain E0432`.
warning: `halp` (lib) generated 1 warning
error: could not compile `halp` (lib) due to 8 previous errors; 1 warning emitted
error: failed to compile `halp v0.2.0 (https://github.com/orhun/halp#17161bbb)`, intermediate artifacts can be found at `/tmp/cargo-installzFCk1C`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

```

--- a/src/helper/docs/cheat_sh.rs
+++ b/src/helper/docs/cheat_sh.rs
@@ -1,6 +1,6 @@
 use crate::error::{Error, Result};
 use crate::helper::docs::HelpProvider;
-use ureq::{AgentBuilder, Request};
+use ureq::Agent;
 
 /// Default cheat sheet provider URL.
 pub const DEFAULT_CHEAT_SHEET_PROVIDER: &str = "https://cheat.sh";
@@ -18,11 +18,16 @@ impl HelpProvider for CheatDotSh {
         DEFAULT_CHEAT_SHEET_PROVIDER
     }
 
-    fn build_request(&self, cmd: &str, url: &str) -> Request {
-        AgentBuilder::new()
+    fn build_request(
+        &self,
+        cmd: &str,
+        url: &str,
+    ) -> ureq::RequestBuilder<ureq::typestate::WithoutBody> {
+        let agent: Agent = Agent::config_builder()
             .user_agent(CHEAT_SHEET_USER_AGENT)
             .build()
-            .get(&format!("{}/{}", url, cmd))
+            .into();
+        agent.get(&format!("{}/{}", url, cmd))
     }
 
     fn fetch(&self, cmd: &str, custom_url: &Option<String>) -> Result<String> {

--- a/src/helper/docs/cheatsheets.rs
+++ b/src/helper/docs/cheatsheets.rs
@@ -1,5 +1,5 @@
 use crate::helper::docs::HelpProvider;
-use ureq::{AgentBuilder, Request};
+use ureq::Agent;
 
 /// The default cheatsheets provider URL.
 pub const DEFAULT_CHEATSHEETS_PROVIDER: &str =
@@ -13,8 +13,13 @@ impl HelpProvider for Cheatsheets {
         DEFAULT_CHEATSHEETS_PROVIDER
     }
 
-    fn build_request(&self, cmd: &str, url: &str) -> Request {
-        AgentBuilder::new().build().get(&format!("{}/{}", url, cmd))
+    fn build_request(
+        &self,
+        cmd: &str,
+        url: &str,
+    ) -> ureq::RequestBuilder<ureq::typestate::WithoutBody> {
+        let agent: Agent = Agent::config_builder().build().into();
+        agent.get(&format!("{}/{}", url, cmd))
     }
 }
 

--- a/src/helper/docs/eg.rs
+++ b/src/helper/docs/eg.rs
@@ -1,5 +1,5 @@
 use crate::helper::docs::HelpProvider;
-use ureq::{AgentBuilder, Request};
+use ureq::Agent;
 
 /// EG page provider URL.
 pub const DEFAULT_EG_PAGES_PROVIDER: &str =
@@ -13,10 +13,13 @@ impl HelpProvider for Eg {
         DEFAULT_EG_PAGES_PROVIDER
     }
 
-    fn build_request(&self, cmd: &str, url: &str) -> Request {
-        AgentBuilder::new()
-            .build()
-            .get(&format!("{}/{}.md", url, cmd))
+    fn build_request(
+        &self,
+        cmd: &str,
+        url: &str,
+    ) -> ureq::RequestBuilder<ureq::typestate::WithoutBody> {
+        let agent: Agent = Agent::config_builder().build().into();
+        agent.get(&format!("{}/{}.md", url, cmd))
     }
 }
 

--- a/src/helper/docs/mod.rs
+++ b/src/helper/docs/mod.rs
@@ -19,8 +19,6 @@ use dialoguer::theme::ColorfulTheme;
 use dialoguer::Select;
 use std::io::Write;
 use std::process::{Command, Stdio};
-use ureq::Request;
-
 /// The `HelpProvider` trait defines essential methods for fetching help content related to commands from a provider.
 ///
 /// Each provider that implements this trait should provide a default URL used to retrieve the command help content.
@@ -55,13 +53,17 @@ pub trait HelpProvider {
     /// - `url`: The root URL.
     ///
     /// # Returns
-    /// This method returns a new `Request` instance configured with the `GET` method and the formatted URL.
-    fn build_request(&self, cmd: &str, url: &str) -> Request;
+    /// This method returns a new `RequestBuilder` configured with the `GET` method and the formatted URL.
+    fn build_request(
+        &self,
+        cmd: &str,
+        url: &str,
+    ) -> ureq::RequestBuilder<ureq::typestate::WithoutBody>;
 
     /// Handle the request error.
     /// aka return a custom message if the error means that **provider** doesn't have a page for the command.
     fn handle_error(&self, e: ureq::Error) -> Error {
-        if e.kind() == ureq::ErrorKind::HTTP {
+        if matches!(e, ureq::Error::StatusCode(_)) {
             Error::ProviderError(
                 "Unknown topic, This topic/command might has no page in this provider yet."
                     .to_string(),
@@ -90,7 +92,10 @@ pub trait HelpProvider {
         let response = response.map_err(|e| self.handle_error(e));
 
         match response {
-            Ok(response) => Ok(response.into_string()?),
+            Ok(mut response) => response
+                .body_mut()
+                .read_to_string()
+                .map_err(|e| Error::from(Box::new(e))),
             Err(e) => Err(e),
         }
     }

